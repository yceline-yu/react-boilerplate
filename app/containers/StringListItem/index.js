/**
 * StringListItem
 *
 * Lists strings
 */

import React from 'react';
import PropTypes from 'prop-types';

import ListItem from 'components/ListItem';
import Wrapper from './Wrapper';

export default function StringListItem(props) {
  const { item } = props;

  // Put together the content
  const content = <Wrapper> {item} </Wrapper>;

  // Render the content into a list item
  return <ListItem key={item} item={content} />;
}

StringListItem.propTypes = {
  item: PropTypes.string,
};
