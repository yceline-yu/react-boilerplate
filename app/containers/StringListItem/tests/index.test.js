/**
 * Test the repo list item
 */

import React from 'react';
import { getByText, render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';

import StringListItem from '../index';

const renderComponent = (props = {}) =>
  render(
    <IntlProvider locale="en">
      <StringListItem {...props} />
    </IntlProvider>,
  );

describe('<StringListItem />', () => {
  let item;

  // Before each test reset the item data for safety
  beforeEach(() => {
    item = 'test';
  });

  it('should render the string', () => {
    const { container } = renderComponent({ item });
    expect(getByText(container, content => content)).not.toBeNull();
  });

  it('should render a ListItem', () => {
    const { container } = renderComponent({ item });
    expect(container.firstChild).toMatchSnapshot();
  });
});
