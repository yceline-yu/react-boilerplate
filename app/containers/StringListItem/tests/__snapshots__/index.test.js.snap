// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StringListItem /> should render a ListItem 1`] = `
<li
  class="Wrapper-euo0oy-0 gpjGxP"
>
  <div
    class="Item-sc-3y9mie-0 jESnYC"
  >
    <div
      class="Wrapper-sc-4ooklm-0 eeXvdc"
    >
       
      test
       
    </div>
  </div>
</li>
`;
